network/tcp_socket.c

```diff
@@ -164,6 +172,28 @@ static int32_t stcp_socket_init(struct secure_socket_desc **desc,
 					  MBEDTLS_SSL_VERIFY_NONE);
 	}
 
+	if (param->cli_cert) {
+		if (!param->cli_pk) {
```

Comment:
this can be moved out of this if statement

Reply:
If a user set cli_cert then it should set also cli_pk. On the other side, if cli_cert is not set then the value of cli_pk doesn't matter. I think checking it outside wouldn't make sense in this context.

Reply:
understood, makes sense.